src/2d/flag2refine2.f90

```diff
@@ -96,8 +96,9 @@ subroutine flag2refine2(mx,my,mbc,mbuff,meqn,maux,xlower,ylower,dx,dy,t,level, &
             ! Consider only snapshots that are within the desired time range
             if (mask_selecta(r)) then
                 ! Calculate inner product with current snapshot
-                call calculate_innerproduct(q,r,mx,my,xlower,ylower,dx, &
-                    dy,meqn,mbc,aux(innerprod_index,1:mx,1:my))
+                call calculate_innerproduct(q,r,mx,my,xlower,   &
+                        ylower,dx,dy,meqn,mbc,aux(1,:,:),       &
```

Comment:
Maybe it would be cleaner to just pass `aux` into `calculate_innerproduct` rather than passing two slices `aux(1,:,:)` and `aux(innerprod_index,1:mx,1:my)`  separately? Then `calculate_innerproduct` can use whatever elements it needs, which might be only `innerprod_index` in many amrclaw applications.  For future extensions some other code (e.g. multi-layer shallow water?) might need more than `aux(1,:,:)` too, so might as well make it all available.  (And I think you can pass the full array with ghost cells as long as it's dimensioned properly in the subroutine.)

Reply:
Sounds like a cleaner approach, I agree. I'll do that.